docs/developer-guide/developer-guide.rst

```diff
@@ -4,50 +4,419 @@
+
+Warm Start
+^^^^^^^^^^
+
+In this mode, the scheduler begins by doing a cold initialization of the driver with its initial settings.
+After initialization, the SchedulerCSC calls the :py:meth:`Driver <lsst.ts.scheduler.driver.Driver.parse_observation_database>`, passing the value of ``startup_database``.
+This method is responsible for reading the input file, in whatever format the specific driver understand and returning a list of :py:class:`Driver <lsst.ts.scheduler.driver_target.DriverTarget>` (or appropriate subclass of).
+The list is then fed back into the :py:class:`Driver <lsst.ts.scheduler.driver.Driver>` by sequentially calling :py:meth:`register_observation <lsst.ts.scheduler.driver.Driver.register_observation>`.
+
+Here the SchedulerCSC assumes that calling :py:meth:`register_observation <lsst.ts.scheduler.driver.Driver.register_observation>` is sufficient to reconstruct the state of a previous observation, without the need to call :py:meth:`select_next_target <lsst.ts.scheduler.driver.Driver.select_next_target>`.
+
+Some implementations may not support this kind of operation.
+In these cases, users must make sure calling :py:meth:`parse_observation_database <lsst.ts.scheduler.driver.Driver.parse_observation_database>` raises an exception to interrupt the SchedulerCSC from attempting to complete the task.
+
+.. _Developer_Guide_Startup_Modes_Hot_Start:
+
+Hot Start
+^^^^^^^^^
+
+In this mode the SchedulerCSC will also begin by doing a cold initialization of the driver with its initial settings.
+Nervertheless, after the initialization process, the SchedulerCSC calls :py:meth:`reset_from_state <lsst.ts.scheduler.driver.Driver.reset_from_state>`, passing the value of ``startup_database``.
+This method should be able to read a snapshot of the state of the scheduler algorithm that reconstruct a previously stored state.
+
+The snapshot must have a similar format to that generated by calling :py:meth:`Driver.save_state <lsst.ts.scheduler.driver.Driver.save_state>`.
+
+This mechanism of saving a snapshot and reloading it afterwards is central to the :ref:`Developer_Guide_Operation_Modes_Advance` scheduler mode of operation.
+
+.. _Developer_Guide_Operation_Modes:
+
+Operation Modes
+---------------
+
+.. _Developer_Guide_Operation_Modes_Simple:
+
+Simple
+^^^^^^
+
+The simple mode of operation causes the scheduler to deal with one target at a time.
+When operating in this mode the Scheduler will;
+
+1. gather telemetry,
+2. update the driver with the most recent telemetry,
+3. request one target,
+4. send the target to the ScriptQueue for execution,
+5. wait for the script execution to complete,
+6. if the observation executes successfully, register the observation in the driver.
+
+This mode is mostly to be used for testing the system is a sequential way.
+
+.. _Developer_Guide_Operation_Modes_Advance:
+
+Advance
+^^^^^^^
+
+The Advance mode is the one intended for actual operation of the observatory.
```

Comment:
In this section, too, `Advaced` would be better than `Advace`. Perhaps change the name of the mode in a future ticket?